.pre-commit-config.yaml

```diff
@@ -96,17 +95,17 @@ repos:
         name: Check for incorrect code block or IPython directives
         language: pygrep
         entry: (\.\. code-block ::|\.\. ipython ::)
-        files: \.(py|pyx|rst)$
+        types_or: [python, cython, rst]
     -   id: unwanted-patterns-strings-to-concatenate
         name: Check for use of not concatenated strings
         language: python
         entry: python scripts/validate_unwanted_patterns.py --validation-type="strings_to_concatenate"
-        files: \.(py|pyx|pxd|pxi)$
+        types_or: [python, file, executable, text]
```

Comment:
I think this will match too many files - it will match anything that is `python` or `file` or `executable` or `text`.

Instead, we only want `\.(py|pyx|pxd|pxi)$` files. `.py` is Python. `.pyx`, `.pxd` and `.pxi` are all Cython:
```bash
(pandas-dev) marco@marco-Predator-PH315-52:~/pandas-dev$ identify-cli pandas/_libs/lib.pxd
["cython", "file", "non-executable", "text"]
(pandas-dev) marco@marco-Predator-PH315-52:~/pandas-dev$ identify-cli pandas/_libs/lib.pyx
["cython", "file", "non-executable", "text"]
(pandas-dev) marco@marco-Predator-PH315-52:~/pandas-dev$ identify-cli ./pandas/_libs/algos_take_helper.pxi
["cython", "executable", "file", "text"]
```

So here we want
```
types_or: [python, cython]
```

Reply:
Done!